rocksdb_replicator/replicated_db.cpp

```diff
@@ -317,7 +331,23 @@ void RocksDBReplicator::ReplicatedDB::pullFromUpstream() {
           }
 
           if (!response.updates.empty()) {
+            db->pullFromUpstreamNoUpdates_ = 0;
             db->cond_var_.notifyAll();
+          } else {
+            incCounter(kReplicatorPullRequestsNoUpdates, 1, db->db_name_);
+            // no updates consecutively, and the upstream says it's NOT a leader.
+            // Therefore we reset upstream. 
+            db->pullFromUpstreamNoUpdates_++;
+            if (response.role != ReplicaRole::LEADER
+                && FLAGS_reset_upstream_on_empty_updates_from_non_leader
```

Comment:
could you elaborate why we need `reset_upstream_on_empty_updates_from_non_leader`? 

when replicated from a nonleader && empty response, shouldn't we reset upstream immediately?

Reply:
This FLAG is for a safe rollout so we don't introduce this change immediately when upgrading each cluster. Once we verified this works for a few clusters, perhaps we can enable this feature by default

> when replicated from a nonleader && empty response, shouldn't we reset upstream immediately?
yep, that's the goal state. Today we want to be a little conservative to avoid overloading Helix/ZK. The plan is to reset immediately (PRs will be coming to achieve this goal, and if you are interested, check the doc I linked in the description).